diff -u -r tigervnc-1.3.1.orig/unix/xserver/hw/vnc/Makefile.am tigervnc-1.3.1/unix/xserver/hw/vnc/Makefile.am
--- tigervnc-1.3.1.orig/unix/xserver/hw/vnc/Makefile.am	2015-01-13 17:27:36.592504774 -0800
+++ tigervnc-1.3.1/unix/xserver/hw/vnc/Makefile.am	2015-01-13 18:11:56.820673316 -0800
@@ -52,7 +52,7 @@
 	-I$(top_srcdir)/hw/xfree86/os-support \
 	-I$(top_srcdir)/hw/xfree86/os-support/bus \
 	-I$(top_srcdir)/include \
-	${XSERVERLIBS_CFLAGS} -I$(includedir)
+	${XSERVERLIBS_CFLAGS}
 
 libvnc_la_LDFLAGS = -module -avoid-version -Wl,-z,now
 
diff -ur tigervnc-1.3.1.orig/unix/xserver/hw/vnc/Makefile.am tigervnc-1.3.1/unix/xserver/hw/vnc/Makefile.am
--- tigervnc-1.3.1.orig/unix/xserver/hw/vnc/Makefile.am	2015-01-13 23:49:34.777485700 -0800
+++ tigervnc-1.3.1/unix/xserver/hw/vnc/Makefile.am	2015-01-13 23:50:14.995454668 -0800
@@ -18,7 +18,7 @@
 
 libvnccommon_la_CPPFLAGS = -DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common -UHAVE_CONFIG_H \
-	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) ${XSERVERLIBS_CFLAGS} -I$(includedir) \
+	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) ${XSERVERLIBS_CFLAGS} \
 	-I$(top_srcdir)/include
 
 bin_PROGRAMS = Xvnc
@@ -35,7 +35,7 @@
 	-UHAVE_CONFIG_H \
 	-DXFree86Server -DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common \
-	-I$(top_srcdir)/include ${XSERVERLIBS_CFLAGS} -I$(includedir)
+	-I$(top_srcdir)/include ${XSERVERLIBS_CFLAGS}
 
 Xvnc_LDADD = $(XVNC_LIBS) libvnccommon.la $(COMMON_LIBS) \
 	$(XSERVER_LIBS) $(XSERVER_SYS_LIBS) $(XVNC_SYS_LIBS) -lX11
diff -u -r tigervnc-1.3.1.orig/unix/xserver/include/misc.h tigervnc-1.3.1/unix/xserver/include/misc.h
--- tigervnc-1.3.1.orig/unix/xserver/include/misc.h	2015-01-13 17:27:36.632502849 -0800
+++ tigervnc-1.3.1/unix/xserver/include/misc.h	2015-01-13 17:44:26.643345836 -0800
@@ -350,7 +350,7 @@
 #define swapl(x) do { \
 		if (sizeof(*(x)) != 4) \
 			wrong_size(); \
-		if (__builtin_constant_p((uintptr_t)(x) & 3) && ((uintptr_t)(x) & 3) == 0) \
+		if (__builtin_constant_p((uint64_t)(x) & 3) && ((uint64_t)(x) & 3) == 0) \
 			*(x) = lswapl(*(x)); \
 		else \
 			swap_uint32((uint32_t *)(x)); \
@@ -369,7 +369,7 @@
 #define swaps(x) do { \
 		if (sizeof(*(x)) != 2) \
 			wrong_size(); \
-		if (__builtin_constant_p((uintptr_t)(x) & 1) && ((uintptr_t)(x) & 1) == 0) \
+		if (__builtin_constant_p((uint64_t)(x) & 1) && ((uint64_t)(x) & 1) == 0) \
 			*(x) = lswaps(*(x)); \
 		else \
 			swap_uint16((uint16_t *)(x)); \
diff -u -r tigervnc-1.3.1.orig/unix/xserver/include/servermd.h tigervnc-1.3.1/unix/xserver/include/servermd.h
--- tigervnc-1.3.1.orig/unix/xserver/include/servermd.h	2015-01-13 17:27:36.632502849 -0800
+++ tigervnc-1.3.1/unix/xserver/include/servermd.h	2015-01-13 17:28:31.979835433 -0800
@@ -300,6 +300,12 @@
 
 #endif                          /* __aarch64__ */
 
+#ifdef __riscv__
+#define IMAGE_BYTE_ORDER        LSBFirst
+#define BITMAP_BIT_ORDER        LSBFirst
+#define GLYPHPADBYTES           4
+#endif
+
 /* size of buffer to use with GetImage, measured in bytes. There's obviously
  * a trade-off between the amount of heap used and the number of times the
  * ddx routine has to be called.
