From 1922e819b923fe6a6d0fe59c6cd819a885a6f1a4 Mon Sep 17 00:00:00 2001
From: Palmer Dabbelt <palmer@dabbelt.com>
Date: Wed, 25 Mar 2015 15:38:31 -0700
Subject: [PATCH] Port to glibc-2.20, which requires _DEFAULT_SOURCE

---
 src/git/date.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/git/date.c b/src/git/date.c
index 8447f45..f8ee4a6 100644
--- a/src/git/date.c
+++ b/src/git/date.c
@@ -5,6 +5,7 @@
  */
 
 #define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 
 #include <gitdate.h>
 #include "date.h"
-- 
2.0.5

