diff -Nur strace-4.8.orig/configure.ac strace-4.8/configure.ac
--- strace-4.8.orig/configure.ac	2015-01-13 11:49:36.746629692 -0800
+++ strace-4.8/configure.ac	2015-01-13 11:50:23.944507772 -0800
@@ -126,6 +126,9 @@
 	arch=xtensa
 	AC_DEFINE([XTENSA], 1, [Define for the Xtensa architecture])
 	;;
+riscv*)
+	arch=riscv
+	AC_DEFINE([RISCV], 1, [Define for RISC-V])
 
 *)
 	AC_MSG_RESULT([NO!])
diff -Nur strace-4.8.orig/configure strace-4.8/configure
--- strace-4.8.orig/configure	2015-01-13 12:00:24.624285234 -0800
+++ strace-4.8/configure	2015-01-13 12:01:14.991814433 -0800
@@ -4782,6 +4782,13 @@
 
 	;;
 
+riscv*)
+	arch=riscv
+
+$as_echo "#define RISCV 1" >>confdefs.h
+
+	;;
+
 *)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: NO!" >&5
 $as_echo "NO!" >&6; }
